scripts/src/osmo-test.ts

```diff
@@ -0,0 +1,632 @@
+    chainID: chainID,
+  });
+  console.log(tx);
+  console.log("----------------------------------");
+  const res = await client.tx.broadcast(tx, chainID);
+  console.log(res);
+  console.log(res.toString());
+  const factoryContract = res.logs[0].events[0].attributes[0].value;
+  console.log("factoryContract", factoryContract);
+  return factoryContract;
+}
+
+async function createAirdrop(
+  wallet: Wallet,
+  factoryContract: string,
+  asset: object,
+  allocated_amounts: number[],
+  from_timestamp: number,
+  to_timestamp: number,
+  campaign_id: string
+) {
+  const createAirdrop = new MsgExecuteContract(
+    wallet.key.accAddress(prefix),
+    factoryContract,
+    {
+      create_airdrop: {
+        asset_info: asset,
+        from_timestamp: from_timestamp,
+        to_timestamp: to_timestamp,
+        allocated_amounts: allocated_amounts.map((x) => x.toString()),
+        campaign_id: campaign_id,
+      },
+    },
+    {}
+  );
+  const tx = await wallet.createAndSignTx({
+    msgs: [createAirdrop],
+    chainID: chainID,
+  });
+  console.log(tx);
+  console.log("----------------------------------");
+  const res = await client.tx.broadcast(tx, chainID);
+  console.log(res);
+  const airdropContract = res.logs[0].events[1].attributes[0].value;
+  console.log("airdropContract", airdropContract);
+  return airdropContract;
+}
+async function increaseIncentives(
+  wallet: Wallet,
+  token_contract: string,
+  amount: number,
+  airdropContract: string
+) {
+  const osmoSend = {
+    send: {
+      contract: airdropContract,
+      amount: amount.toString(),
+      msg: Buffer.from(
+        JSON.stringify({
+          increase_incentives: {},
+        })
+      ).toString("base64"),
+    },
+  };
+  const sendTokens = new MsgExecuteContract(
+    wallet.key.accAddress(prefix),
+    token_contract,
+    osmoSend,
+    {}
+  );
+
+  const tx = await wallet.createAndSignTx({
+    msgs: [sendTokens],
+    chainID: chainID,
+  });
+  console.log(tx);
+  console.log("----------------------------------");
+  const res = await client.tx.broadcast(tx, chainID);
+  console.log(res);
+}
+
+// async function topupIncentives(
+//   wallet: Wallet,
+//   token_contract: string,
+//   amount: number[],
+//   airdropContract: string
+// ) {
+//   const osmoSend = {
+//     send: {
+//       contract: airdropContract,
+//       amount: amount.toString(),
+//       msg: Buffer.from(
+//         JSON.stringify({
+//           topup_incentives: {},
+//         })
+//       ).toString("base64"),
+//     },
+//   };
+//   const sendTokens = new MsgExecuteContract(
+//     wallet.key.accAddress(prefix),
+//     token_contract,
+//     osmoSend,
+//     {}
+//   );
+
+//   const tx = await wallet.createAndSignTx({
+//     msgs: [sendTokens],
+//     chainID: chainID,
+//   });
+//   console.log(tx);
+//   console.log("----------------------------------");
+//   const res = await client.tx.broadcast(tx, chainID);
+//   console.log(res);
+// }
```

Comment:
Can we delete this if its not neeeded

Reply:
I understand what you mean. Should have deleted it, but i leave it in for now so I can uncomment it later.